### PR TITLE
[REEF-360]: Fix violations / reconfigure checkstyle checks in "Blocks" category

### PR DESCRIPTION
**Pull request by tcNickolas**
This addressed the issue by
  * fixing LeftCurly and NeedBraces violations
  * fixing most AvoidNestedBlocks and EmptyBlock violations
  * configuring AvoidNestedBlocks to allow nested blocks in switch case
  * configuring EmptyBlock to allow blocks with comments only
  * allowing to configure per-case warning suppressions and suppressing AvoidNestedBlocks in org/apache/reef/util/MemoryUtils.java

JIRA: [REEF-360](https://issues.apache.org/jira/browse/REEF-360)

### CODE REVIEW

File: lang/java/reef-common/src/main/java/org/apache/reef/util/MemoryUtils.java [hunk 1]
```diff
@@ -96,6 +96,7 @@ public static void resetPeakUsage() {
     }
   }
 
+  @SuppressWarnings("checkstyle:avoidnestedblocks")
```
Comment (dafrista) [inline lang/java/reef-common/src/main/java/org/apache/reef/util/MemoryUtils.java]: It looks like the nested block can safely be removed, and then we won't want this suppressed.

File: lang/java/reef-io/src/main/java/org/apache/reef/io/data/loading/impl/NumberedSplit.java [hunk 2]
```diff
@@ -53,11 +53,14 @@ public String toString() {
...
-    if (this.index == o.index)
+    if (this.index == o.index) {
       return 0;
-    if (this.index < o.index)
+    }
+    if (this.index < o.index) {
       return -1;
-    else
+    }
+    else {
```
Comment (dafrista) [inline lang/java/reef-io/src/main/java/org/apache/reef/io/data/loading/impl/NumberedSplit.java]: should be on the same line `} else {`

**Comment by dafrista** [comment]
@tcNickolas I made a pass and commented on a couple parts. +1 once these are looked at.

File: lang/java/reef-common/src/main/java/org/apache/reef/io/Tuple.java [hunk 3]
```diff
@@ -41,13 +41,21 @@ public V getValue() {
 
   @Override
   public boolean equals(final Object o) {
```
Comment (markusweimer) [inline lang/java/reef-common/src/main/java/org/apache/reef/io/Tuple.java]: The sad part is that this is auto-generated code in most cases. Hopefully we can convince IntelliJ to generate code that complies with CheckStyle in the future.

File: lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/DriverStatusManager.java [hunk 4]
```diff
@@ -180,17 +180,17 @@ public synchronized void sendJobEndingMessageToClient(final Optional<Throwable>
...
-        if (this.shutdownCause.isPresent()) {
-          LOG.log(Level.WARNING, "Sending message about an unclean driver shutdown.", this.shutdownCause.get());
-        }
-        if (exception.isPresent()) {
-          LOG.log(Level.WARNING, "There was an exception during clock.close().", exception.get());
-        }
-        if (this.shutdownCause.isPresent() && exception.isPresent()) {
-          LOG.log(Level.WARNING, "The driver is shutdown because of an exception (see above) and there was an exception during clock.close(). Only the first exception will be sent to the client");
-        }
+      // Log the shutdown situation
```
Comment (markusweimer) [inline lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/DriverStatusManager.java]: This seems like a pure formatting change, right? If so, please remove it from the PR.

**Comment by markusweimer** [comment]
I also did a pass.

File: lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/DriverStatusManager.java (hunk 4 above)
Comment (tcNickolas) [inline lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/DriverStatusManager.java]: This removed an unnecessary nested block ({ at line 183 and matching } at line 193) and adjusted indentation accordingly.

Comment (markusweimer) [inline lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/DriverStatusManager.java]: Thanks!

**Comment by dafrista** [comment]
Thanks @markusweimer for the second pass. I'll test and merge.

File: lang/java/reef-common/src/main/java/org/apache/reef/io/Tuple.java (hunk 3 above)
Comment (dafrista) [inline lang/java/reef-common/src/main/java/org/apache/reef/io/Tuple.java]: @markusweimer I tried out a few things and figured out that IntelliJ code generation actually follows the code style configuration. So the thing to set is `Editor > Code Style > Java > Wrapping and Braces > if() statement > Force braces > Always`, and then the generation will add if braces.

**Pull request closed without merging**